components/sup/src/manager/service/config.rs

```diff
@@ -60,30 +63,71 @@ pub struct ServiceConfig {
     // Set to 'true' if we have data that needs to be sent to a configuration file
     #[serde(skip_deserializing)]
     pub needs_write: bool,
+    #[serde(skip_deserializing)]
+    supported_bindings: Vec<(String, String)>,
```

Comment:
Perhaps a `Vec<(String, ServiceConfig)>` instead? I was hoping to validate this input a lot earlier in the Supervisor's lifecyle and this might help.